### ISSUE TITLE
`/metadata/json:validate` endpoint returns HTTP 500 error in Napa and prod (not dev)

### ISSUE THREAD

**Issue by aclum** [microbiomedata/nmdc-runtime#524]
The json:validate endpoint is having issues
Working:
curl call to json:validate on dev
Swagger UI  on dev not tested
Not Working:
swagger UI or curl on napa
swagger UI or curl on prod

Error: "Internal Server Error"

this is a blocker for testing of re-iding.

cc @shreddd @dwinston @eecavanna

**Comment by aclum** [comment]
cc @mbthornton-lbl

**Comment by eecavanna** [comment]
I wonder whether the new, user-defined role I assigned to the user that the Runtime connects to Mongo as, doesn't allow it to do _whatever it does_ with the `json:validate` endpoint (for reference, the Runtime used to connect to Mongo as the _root_ user). Note that I have applied those changes to the production and Napa environments (I applied them to the Napa environment just last night), but not the _dev_ environment. I will check once I'm back home, which will be by 8pm.

**Comment by eecavanna** [comment]
Here are some specifics, gathered using the Swagger UI of the Napa environment:

When the payload is `{}`, the endpoint returns:
```json
{
  "result": "All Okay!"
}
```

So, the endpoint is able to do the bare minimum type of validation.

(...more to come...)

**Comment by eecavanna** [comment]
@aclum or @mbthornton-lbl, I haven't been able to reproduce the issue. Will you share a problematic request body (a.k.a. request payload) with me?

**Comment by eecavanna** [comment]
1. I wasn't able to reproduce the issue (the request bodies I came up with didn't trigger an HTTP 500 response).
2. Based on the stack trace I see in the server logs (which I assume are associated with this HTTP requests about which this ticket was created), it does look to me like the Mongo user the Runtime uses to access the Mongo server, doesn't have permission to do whatever the `/metadata/json:validate` endpoint tries to do when the problematic requests are submitted.
3. Just now, in the Napa environment, I granted additional roles to the Mongo user the Runtime uses to access the Mongo server. Those additional roles are `readWriteAnyDatabase` and `dbAdminAnyDatabase`.
4. Just now, in the production environment, I granted an additional role to the Mongo user the Runtime uses to access the Mongo server. That additional role is `root` (just in case the two roles mentioned in the previous bullet point are insufficient) (I'll eventually standardize the role setup across both environments).

**Comment by eecavanna** [comment]
Things I want to know (@aclum and/or @mbthornton-lbl):

1. Now that I've made the above change (in bullet point 3 above): Is the same request that was failing (getting an HTTP 500 response) in the Napa environment, still failing in the Napa environment?
2. Now that I've made the above change (in bullet point 4 above): Is the same request that was failing (getting an HTTP 500 response) in the production environment, still failing in the production environment?
3. What is an example failing request body (for my future reference)?
